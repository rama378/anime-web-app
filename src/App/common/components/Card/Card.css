.Card {
  cursor: pointer;
  background-color: #212121;
  border-radius: 8px;
  padding: 1px;
  transition: 0.3s all;
}

.Card:hover {
  background-color: #2e2e2e;
  transition: 0.3s all;
}

.Card__wrapper {
  padding: 8px;
}

.Card__imagewrapper {
  border-radius: 8px;
  margin: 16px;
  max-height: 340px;
  overflow: hidden;
}

.Card__image {
  width: 100%;
}

.Card__title {
  margin: 8px 16px 16px 16px;
  font-weight: bold;
  font-size: var(--font-size-large);
}

.Card__footer {
  margin: 0 16px 16px 16px;
}

.Card__ratingicon {
  width: 20px;
  height: 20px;
  fill: #fff;
  vertical-align: middle;
  margin-right: 4px;
}

.Card__rating {
  font-size: var(--font-size-small);
  margin-top: 8px;
}

.Card__ratinglabel {
  opacity: 0.5;
}
