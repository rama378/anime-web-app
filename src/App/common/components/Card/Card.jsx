import PropTypes from 'prop-types';
import './Card.css';
import {ReactComponent as UserRatingIcon} from "../../assets/images/icon/star-comment-alt.svg";
import { formatNumber } from '../../utils/numberUtils';
import { PAGE } from '../../constants/constant';
import { useNavigate } from 'react-router-dom';

const Card = (props) => {
	const {detail, className, ...rest} = props;

	const navigate = useNavigate();
    
	const openDetail = () => {
			navigate(`${PAGE.DETAIL}/${detail?.mal_id}`);
	};

	return (
		<div className={`Card__wrapper ${className}`} onClick={openDetail}>
			<div className='Card' {...rest}>
				<div className='Card__imagewrapper'>
					<img src={detail?.images?.webp?.image_url} className='Card__image' />
				</div>

				<div className='Card__title'>{detail?.title}</div>

				<div className='Card__footer'>
					<div>
						<UserRatingIcon className='Card__ratingicon' /> {detail?.score} ({formatNumber(detail?.scored_by)})
					</div>
					<div className='Card__rating'>
						<span className='Card__ratinglabel'>Rating : </span> {detail?.rating}
					</div>
				</div>
			</div>
		</div>
	);
};

Card.propTypes = {
	detail: PropTypes.object.isRequired,
	className: PropTypes.string,
};

Card.defaultProps = {
	className: null,
};

export default Card;
