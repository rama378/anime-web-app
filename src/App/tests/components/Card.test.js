import React from 'react';
import { BrowserRouter } from 'react-router-dom';
import ShallowRenderer from 'react-test-renderer/shallow';

import Card from '../../common/components/Card/Card';

const renderer = new ShallowRenderer();

describe('<Card />', () => {
  it('should render and match the snapshot given detail prop', () => {
    renderer.render((
      <BrowserRouter>
        <Card detail={{
          mal_id: 1,
          url: 'https://myanimelist.net/anime/1/Cowboy_Bebop',
          images: {
            jpg: {
              image_url: 'https://cdn.myanimelist.net/images/anime/4/19644.jpg',
            },
            webp: {
              image_url: 'https://cdn.myanimelist.net/images/anime/4/19644.webp',
            },
          },
          title: 'Cowboy Bebop',
          score: 1.5,
          scored_by: 12345,
          rating: 'PG-13 - Teens 13 or older',
        }} />
      </BrowserRouter>
    ));
    const renderedOutput = renderer.getRenderOutput();
    expect(renderedOutput).toMatchSnapshot();
  });
});
