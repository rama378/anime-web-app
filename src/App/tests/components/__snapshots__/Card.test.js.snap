// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Card /> should render and match the snapshot given detail prop 1`] = `
<Router
  location={
    Object {
      "hash": "",
      "key": "default",
      "pathname": "/",
      "search": "",
      "state": null,
    }
  }
  navigationType="POP"
  navigator={
    Object {
      "action": "POP",
      "createHref": [Function],
      "encodeLocation": [Function],
      "go": [Function],
      "listen": [Function],
      "location": Object {
        "hash": "",
        "key": "default",
        "pathname": "/",
        "search": "",
        "state": null,
      },
      "push": [Function],
      "replace": [Function],
    }
  }
>
  <Card
    className={null}
    detail={
      Object {
        "images": Object {
          "jpg": Object {
            "image_url": "https://cdn.myanimelist.net/images/anime/4/19644.jpg",
          },
          "webp": Object {
            "image_url": "https://cdn.myanimelist.net/images/anime/4/19644.webp",
          },
        },
        "mal_id": 1,
        "rating": "PG-13 - Teens 13 or older",
        "score": 1.5,
        "scored_by": 12345,
        "title": "Cowboy Bebop",
        "url": "https://myanimelist.net/anime/1/Cowboy_Bebop",
      }
    }
  />
</Router>
`;
